pkg/theiactl/portforwarder/portforwarder.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package portforwarder
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+type PortForwarder struct {
+	Config        *rest.Config
+	Clientset     kubernetes.Interface
+	Namespace     string
+	Name          string
+	TargetPort    int
+	ListenAddress string
+	ListenPort    int
+	stopCh        chan struct{}
+}
+
+// This function creates Port Forwarder for a Pod
+// After creating Port Forwarder object, call Start() on it to start forwarding
+// channel and Stop() to terminate it
+func NewPortForwarder(config *rest.Config, namespace string, pod string, listenAddress string, listenPort int, targetPort int) (*PortForwarder, error) {
```

Comment:
Could we change the parameter order to something like "pod string, targetPort int, listenAddress string, listenPort int"? The current order seems a little bit strange.

Reply:
I was following the order for `kubectl port-forward` command, but your suggestion makes more sense to me too, so I changed the order. Thanks!